packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1029,36 +1033,56 @@ const clientInitializationKey = createStateSymbol("clientInitialization");
 export const $clientInitialization: ClientInitializationDecorator = (
   context: DecoratorContext,
   target: Namespace | Interface,
-  options: Model,
+  options: Type,
   scope?: LanguageScopes,
 ) => {
-  setScopedDecoratorData(
-    context,
-    $clientInitialization,
-    clientInitializationKey,
-    target,
-    options,
-    scope,
-  );
+  if (options.kind === "Model") {
+    if (options.properties.get("access")) {
+      validateAccessValue(context, target, options.properties.get("access")!.type as EnumMember);
+    }
+    if (options.properties.get("accessorAccess")) {
+      validateAccessValue(
+        context,
+        target,
+        options.properties.get("accessorAccess")!.type as EnumMember,
+      );
+    }
+
+    setScopedDecoratorData(
+      context,
+      $clientInitialization,
+      clientInitializationKey,
+      target,
+      options,
+      scope,
+    );
+  }
 };
 
 export function getClientInitialization(
   context: TCGCContext,
   entity: Namespace | Interface,
-): SdkInitializationType | undefined {
-  const model = getScopedDecoratorData(context, clientInitializationKey, entity);
-  if (!model) return model;
-  const sdkModel = getSdkModel(context, model);
-  const initializationProps = sdkModel.properties.map(
-    (property: SdkModelPropertyType): SdkMethodParameter => {
-      property.onClient = true;
-      property.kind = "method";
-      return property as SdkMethodParameter;
-    },
-  );
+): ClientInitializationOptions | undefined {
+  const options = getScopedDecoratorData(context, clientInitializationKey, entity);
+  if (options === undefined) return undefined;
+
+  const model = options.properties.get("parameters")
+    ? getSdkModel(context, options.properties.get("parameters").type)
+    : undefined;
   return {
-    ...sdkModel,
-    properties: initializationProps,
+    parameters: model
+      ? model.properties.map((property: SdkModelPropertyType): SdkMethodParameter => {
+          property.onClient = true;
+          property.kind = "method";
+          return property as SdkMethodParameter;
+        })
+      : undefined,
+    access: options.properties.get("access")
+      ? options.properties.get("access").type.value
```

Comment:
nit: I think you can do `options.properties.get('access')?.type?.value || undefined`, or maybe you don't even need `|| undefined`